Add file filter to remove files with tilde suffix

madkins23 opened the pull request:
On my platform (Ubuntu Linux) any file I edit with DrRacket gets duplicated with a tilde (`~`) at the end, for example:

```
$ ls -l tool.*
-rw-rw-r-- 1 marc marc 16188 Jun 17 20:55 tool.rkt
-rw-rw-r-- 1 marc marc 16188 Jun 17 20:47 tool.rkt~
```

I've added to the filter dialog and preferences a choice to filter out these files.

Since I don't have a Mac or Windows box I don't know if this is a common affliction. If not, this code should probably be extended to avoid the choice for other platforms. On the other hand, it shouldn't break anything and the prefix period preference probably only applies to Linux so maybe it doesn't matter.

Syntacticlosure commented:
In Windows, DrRacket backup files have .bak suffixes.
In Linux/Mac, they have ~ suffixes.
Providing an additional option to hide these files makes sense even though it could be done by setting a ".bak ~" filter.

I think u can generalize this option to work in different operating systems.

madkins23 commented:
I thought I had tried a `~` filter earlier and it didn't work but I tried it just now and it does. Maybe what I actually tested was `*~` which isn't going to work the way the code is written (just checking suffix, not a regex match).

If it had worked for me when I tried it I wouldn't have submitted this PR. Good way to get started with the code, I guess. ;-)

I've updated the PR to be OS sensitive and configured it per you comment regarding suffixes on different platforms. I don't have any way to test Mac and Windows. I've also changed the filter to say "backup files" with the suffix to be removed provided by OS.

madkins23 commented on private/gui-helpers.rkt:
```diff
@@ -204,7 +209,10 @@
                                  (ormap (λ (x) (path-has-extension? i x)) filter-types))))
-                   (not (and (preferences:get 'files-viewer:filter-types3) (string-prefix? (path->string i) ".")))
+                   (not (and (preferences:get 'files-viewer:filter-types3)
+                             (string-prefix? (path->string i) ".")))
```
I've got the editor preference for the 102 character vertical line turned on. Mine below was way over and the one above was a bit over so I broke both of them. I'm not going to go through and do style fixes as a general rule, not my code. Just figured I would explain the change.

Syntacticlosure commented:
LGTM, I will test it later.
